test/sched.py

```diff
@@ -53,13 +55,20 @@
 
 def set_PBS_env():
     """ Set up the environment to recreate being in a hpc job """
-    pbsnodefile = tempfile.NamedTemporaryFile(delete=False)
+    if not os.path.isdir('/tmp/%s' % os.environ['USER']):
+        os.mkdir('/tmp/%s' % os.environ['USER'])
+    pbsnodefile = tempfile.NamedTemporaryFile(dir='/tmp/%s' % os.environ['USER'], delete=False)
     pbsnodefile.write("localhost\nlocalhost\n")
     pbsnodefile.close()
     os.environ['PBS_NODEFILE'] = pbsnodefile.name
+    os.chmod(pbsnodefile.name, stat.S_IRUSR)
```

Comment:
include a comment to explain why we do this (i.e., in a real situation the file and location of `$PBS_NODEFILE` are not writeable either)